src/services/member/plugins/action/service.ts

```diff
@@ -1,15 +1,54 @@
-import { CannotModifyOtherMembers } from '../../../../utils/errors';
+import { ActionTriggers } from '@graasp/sdk';
+
+import { CannotModifyOtherMembers, UnauthorizedMember } from '../../../../utils/errors';
 import { Repositories } from '../../../../utils/repositories';
 import { ActionService } from '../../../action/services/action';
+import { ItemActionType } from '../../../item/plugins/action/utils';
 import { Actor } from '../../entities/member';
 
+export const getMonthBeforeNow = () => {
+  const date = new Date(); // Today's date
+  date.setMonth(date.getMonth() - 1); // Set the date to one month ago
+  return date;
+};
+
 export class ActionMemberService {
   actionService: ActionService;
 
   constructor(actionService: ActionService) {
     this.actionService = actionService;
   }
 
+  async getFilteredActions(
+    actor: Actor,
+    repositories: Repositories,
+    filters: { startDate?: string; endDate?: string },
+  ) {
+    const { actionRepository } = repositories;
+
+    const { startDate, endDate } = filters;
+    const start = startDate ? new Date(startDate) : getMonthBeforeNow();
+    const end = endDate ? new Date(endDate) : new Date();
+
+    const allowedTypes = [
+      ActionTriggers.MemberLogin,
+      ItemActionType.Create,
+      ItemActionType.Update,
+      ItemActionType.Delete,
+    ];
+
+    if (!actor) {
+      throw new UnauthorizedMember(actor);
+    }
+    const data = await actionRepository.getForMember(actor.id, {
```

Comment:
I've realised that you might have actions for item you don't have access to anymore. I think it's safer if you run `validatePermissionMany` over all the items (a set of them, since they will exist multiple times in the actions). Also add a comment because that might not be the most efficient way to get the right set of actions... if you can think of another way don't hesitate. You can find an example of what I have in mind in `getParents` on the item service.